ports/esp32/network_ppp.c

```diff
@@ -144,19 +142,16 @@ static mp_obj_t ppp_active(size_t n_args, const mp_obj_t *args) {
             }
 
             if (self->client_task_handle != NULL) { // is connecting or connected?
-                // Wait for PPPERR_USER, with timeout
-                pppapi_close(self->pcb, 0);
-                uint32_t t0 = mp_hal_ticks_ms();
-                while (!self->clean_close && mp_hal_ticks_ms() - t0 < PPP_CLOSE_TIMEOUT_MS) {
-                    mp_hal_delay_ms(10);
-                }
+                // Stop task
+                vTaskDelete(self->client_task_handle);
+                self->client_task_handle = NULL;
+            }
 
-                // Shutdown task
-                xTaskNotifyGive(self->client_task_handle);
-                t0 = mp_hal_ticks_ms();
-                while (self->client_task_handle != NULL && mp_hal_ticks_ms() - t0 < PPP_CLOSE_TIMEOUT_MS) {
-                    mp_hal_delay_ms(10);
-                }
+            // Close ppp connection
+            pppapi_close(self->pcb, 0);
+            while (!self->clean_close) {
+                // Wait for clean close
+                mp_hal_delay_ms(10);
```

Comment:
I'm not sure what you've done here is correct.  With the changes here it will first delete the `pppos_client_task` task and then attempt to close the PPP connection.  But `pppos_client_task` needs to remain active to be able to forward data packets from the UART into the PPP driver, so it can cleanly shut down.

The way it's done at the moment (close PPP then stop task) seems correct.